pandas/core/computation/eval.py

```diff
@@ -39,10 +39,10 @@ def _check_engine(engine: Optional[str]) -> str:
     str
         Engine name.
     """
-    from pandas.core.computation.check import _NUMEXPR_INSTALLED
+    from pandas.core.computation.expressions import _NUMEXPR_INSTALLED, _USE_NUMEXPR
 
     if engine is None:
-        engine = "numexpr" if _NUMEXPR_INSTALLED else "python"
+        engine = "numexpr" if _USE_NUMEXPR else "python"
```

Comment:
i think in this case we need to check `expr.USE_NUMEXPR`, _not_ `USE_NUMEXPR` since the former may change at runtime